src/schrodinger/rdkit_extensions/helm/monomer_coordgen.cpp

```diff
@@ -307,17 +302,18 @@ static std::vector<int> find_largest_ring(const RDKit::ROMol& polymer)
         RDKit::MolOps::findSSSR(polymer, /*res=*/nullptr, include_dative_bonds);
     }
 
+    std::vector<int> largest_ring_no_connections{};
     std::vector<int> largest_ring{};
     for (auto cycle : polymer.getRingInfo()->atomRings()) {
-        if (cycle.size() <= largest_ring.size()) {
+        if (cycle.size() <= largest_ring_no_connections.size()) {
             continue;
         }
-
         std::vector<unsigned int> connection_points{};
         for (size_t i = 0; i < cycle.size(); i++) {
             auto monomer = cycle[i];
             auto next_monomer = i == cycle.size() - 1 ? cycle[0] : cycle[i + 1];
             auto bond = polymer.getBondBetweenAtoms(monomer, next_monomer);
+
             if (bond->hasProp(CUSTOM_BOND)) {
```

Comment:
Do we care if this custom bond is to another member of the ring versus something outside of the ring?